### ISSUE TITLE
Issue with Agenda 'this.props.renderItem is not a function'

### ISSUE THREAD

**Issue by Jessenw** [wix/react-native-calendars#935]
## Description

I am currently receiving an error when I click on an item in Agenda.

## Expected Behavior

Expectation is that it shows the list of items as shown in the Agenda documentation

## Observed Behavior

See error output

```
ExceptionsManager.js:94 Unhandled JS Exception: TypeError: TypeError: this.props.renderItem is not a function

This error is located at:
    in IGNORE (at reservation-list/index.js:117)
    in RCTView (at View.js:35)
    in View (at reservation-list/index.js:116)
    in RCTView (at View.js:35)
    in View (at VirtualizedList.js:1770)
    in CellRenderer (at VirtualizedList.js:688)
    in RCTScrollContentView (at ScrollView.js:1007)
    in RCTScrollView (at ScrollView.js:1147)
    in ScrollView (at VirtualizedList.js:1081)
    in VirtualizedList (at FlatList.js:634)
    in FlatList (at reservation-list/index.js:195)
    in IGNORE (at agenda/index.js:284)
    in RCTView (at View.js:35)
    in View (at agenda/index.js:399)
    in RCTView (at View.js:35)
    in View (at agenda/index.js:398)
    in Agenda (at CalendarScreen.js:18)
    in RCTView (at View.js:35)
    in View (at Container.js:12)
    in Container (at connectStyle.js:392)
    in Styled(Container) (at CalendarScreen.js:17)
    in CalendarScreen (at SceneView.js:9)
    in SceneView (at StackViewLayout.tsx:909)
    in RCTView (at View.js:35)
    in View (at createAnimatedComponent.js:151)
    in AnimatedComponent (at StackViewCard.tsx:106)
    in RCTView (at View.js:35)
    in View (at createAnimatedComponent.js:151)
    in AnimatedComponent (at screens.native.js:94)
    in Screen (at StackViewCard.tsx:93)
    in Card (at createPointerEventsContainer.tsx:95)
    in Container (at StackViewLayout.tsx:984)
    in RCTView (at View.js:35)
    in View (at screens.native.js:124)
    in ScreenContainer (at StackViewLayout.tsx:393)
    in RCTView (at View.js:35)
    in View (at createAnimatedComponent.js:151)
    in AnimatedComponent (at StackViewLayout.tsx:383)
    in PanGestureHandler (at StackViewLayout.tsx:376)
    in StackViewLayout (at withOrientation.js:30)
    in withOrientation (at StackView.tsx:104)
    in RCTView (at View.js:35)
    in View (at Transitioner.tsx:267)
    in Transitioner (at StackView.tsx:41)
    in StackView (at createNavigator.js:80)
    in Navigator (at createKeyboardAwareNavigator.js:12)
    in KeyboardAwareNavigator (at SceneView.js:9)
    in SceneView (at SwitchView.js:12)
    in SwitchView (at createNavigator.js:80)
    in Navigator (at createAppContainer.js:430)
    in NavigationContainer (at renderApplication.js:40)
    in RCTView (at View.js:35)
    in View (at AppContainer.js:98)
    in RCTView (at View.js:35)
    in View (at AppContainer.js:115)
    in AppContainer (at renderApplication.js:39)

This error is located at:
    in NavigationContainer (at renderApplication.js:40)
    in RCTView (at View.js:35)
    in View (at AppContainer.js:98)
    in RCTView (at View.js:35)
    in View (at AppContainer.js:115)
    in AppContainer (at renderApplication.js:39)
```

## Environment

Please run these commands in the project folder and fill in their results:

* `npm ls react-native-calendars`: 
```
StudentCalendar@0.0.1 /Users/jessenw/Desktop/student-calendar-react-native/StudentCalendar
└── react-native-calendars@1.212.0 
```
* `npm ls react-native`:
```
StudentCalendar@0.0.1 /Users/jessenw/Desktop/student-calendar-react-native/StudentCalendar
└── react-native@0.60.5 
```

Also specify:

1. Phone/emulator/simulator & version: Running on iOS simulator

## Reproducible Demo

Please provide a minimized reproducible demonstration of the problem you're reporting.

```
import React, { Component } from 'react';
import { Calendar, Agenda } from "react-native-calendars";

export default class CalendarScreen extends Component {
    static navigationOptions = {
        title: 'Calendar',
    };

    render() {
        return (
            <Agenda
                items={{
                    '2019-09-23': [{text: 'item 1 - any js object'}],
                    '2019-09-24': [{text: 'item 2 - any js object'}]
                }}
            />
        );
    }
}
```

**Comment by Bright24** [comment]
You aren't passing the renderItem prop which throws the error I would think. 

See the official docs for the agenda example...
https://github.com/wix/react-native-calendars/tree/master/example/src/screens

**Comment by Inbal-Tish** [comment]
@Jessenw First, `renderItem` and `rowHasChanged` props are required. I will mark them as such to avoid confusion. 
Second, you might want to add `selected={'2019-09-23'}` since the current date (the default) has no items and will show an activity indicator until you select one of the dates in your list of items. If you add this prop, it will just set on one of your items and load it.

**Comment by Inbal-Tish** [comment]
@Jessenw See fix on version 1.257.0
